app/src/main/java/com/framgia/fsalon/utils/formatter/CustomMarkerView.java

```diff
@@ -0,0 +1,43 @@
+
+package com.framgia.fsalon.utils.formatter;
+
+import android.content.Context;
+import android.widget.TextView;
+
+import com.framgia.fsalon.R;
+import com.github.mikephil.charting.components.MarkerView;
+import com.github.mikephil.charting.data.CandleEntry;
+import com.github.mikephil.charting.data.Entry;
+import com.github.mikephil.charting.highlight.Highlight;
+import com.github.mikephil.charting.utils.MPPointF;
+import com.github.mikephil.charting.utils.Utils;
+
+/**
+ * Custom implementation of the MarkerView.
+ *
+ * @author Philipp Jahoda
+ */
+public class CustomMarkerView extends MarkerView {
+    private TextView mTextContent;
+
+    public CustomMarkerView(Context context, int layoutResource) {
+        super(context, layoutResource);
+        mTextContent = (TextView) findViewById(R.id.text_content);
+    }
+
+    @Override
+    public void refreshContent(Entry e, Highlight highlight) {
+        if (e instanceof CandleEntry) {
+            CandleEntry ce = (CandleEntry) e;
+            mTextContent.setText("" + Utils.formatNumber(ce.getHigh(), 0, true));
+        } else {
+            mTextContent.setText("" + Utils.formatNumber(e.getY(), 0, true));
```

Comment:
WARNING: [SetTextI18n] Do not concatenate text displayed with `setText`. Use resource string with placeholders.
 When calling `TextView#setText`
* Never call `Number#toString()` to format numbers; it will not handle fraction separators and locale-specific digits properly. Consider using `String#format` with proper format specifications (`%d` or `%f`) instead.
* Do not pass a string literal (e.g. "Hello") to display text. Hardcoded text can not be properly translated to other languages. Consider using Android resource strings instead.
* Do not build messages by concatenating text chunks. Such messages can not be properly translated.